MBS-8206: Add Disc ID report

naiveaiguy opened the pull request:
This report shows all the releases that have at least one medium that has a format with the `has_discids` property, but that don't actually have a Disc ID attached to it.

naiveaiguy commented:
I've just noticed that [MBS-8206](https://tickets.metabrainz.org/browse/MBS-8206) specifies that we should ignore psedo and bootleg releases, but the GCI task I am coming from specifies to only list pseudo-releases. The latter feels like a mistake to me, so I'm going to amend the commit using the former specification.

yvanzo commented on lib/MusicBrainz/Server/Report/ReleasesWithoutDiscIds.pm:
```diff
@@ -0,0 +1,45 @@
+          FROM release r
+          WHERE id IN (
+              SELECT release FROM medium
+              WHERE (SELECT has_discids FROM medium_format WHERE id = format)
```
I would use `medium_format.id = medium.format` to make it less ambiguous.

yvanzo commented on lib/MusicBrainz/Server/Report/ReleasesWithoutDiscIds.pm:
```diff
@@ -0,0 +1,45 @@
+__PACKAGE__->meta->make_immutable;
+1;
+
+=head1 COPYRIGHT
```
Please use the lighter and up-to-date section [`COPYRIGHT AND LICENSE`](https://musicbrainz.org/doc/Development/Server_Coding_Style#Perl_Module_Template_and_Documentation_Guidelines) instead.

yvanzo commented on root/report/index.tt:
```diff
@@ -108,6 +108,7 @@
             <li><a href="[% c.uri_for_action('/report/show', 'ReleasesWithNoMediums') %]">[% l('Releases with no mediums') %]</a></li>
             <li><a href="[% c.uri_for_action('/report/show', 'ReleasesWithoutVACredit') %]">[% l('Releases not credited to "Various Artists" but linked to VA') %]</a></li>
             <li><a href="[% c.uri_for_action('/report/show', 'ReleasesWithoutVALink') %]">[% l('Releases credited to "Various Artists" but not linked to VA') %]</a></li>
+            <li><a href="[% c.uri_for_action('/report/show', 'ReleasesWithoutDiscIds') %]">[% l('Releases without Disc IDs') %]</a></li>
```
The URL is fine, but the title is a bit confusing. It is legitimate to have (e.g. Vinyl) releases without Disc IDs. The title should explain the weirdness of reported releases. Replacing _without_ by _missing_ will probably do.

yvanzo commented on root/report/releases_without_disc_ids.tt:
```diff
@@ -0,0 +1,16 @@
+
+<ul>
+  <li>[% l('This report shows releases (official and promotional only) that have at least
+            one medium without a Disc ID. For instructions on how to add one, see the {add_discids|documentation page}.',
```
The same remark as for the title. Maybe a bit more detailed explanation would clarify it.

naiveaiguy commented:
Am here. On it.

naiveaiguy commented:
Amended commit with changes. Note, when I added the line to ignore pseudo and bootleg releases earlier, I made the grave mistake of not testing it. Now, when I was, I saw that the query takes absolutely **forever** with that line. For now, I have removed it.

Freso commented on lib/MusicBrainz/Server/Report/ReleasesMissingDiscIds.pm:
```diff
@@ -0,0 +1,35 @@
+
+=head1 COPYRIGHT
+
+Copyright (C) 2017 MetaBrainz Foundation
```
You should use your own name here. MetaBrainz is not paying for your work, so you retain your copyright. :)

naiveaiguy replied:
Well I'd like to think I'm part of the team 😸

yvanzo replied:
Actually, I did the same as a free contributor in the past. Giving the copyright to the MetaBrainz Foundation is mostly about [_procedural advantages_ [...] _under US copyright law_](https://www.gnu.org/licenses/why-assign.html). Copyright in your own name would be accepted too.

naiveaiguy replied:
Alright. Will do in #588 as well.

yvanzo commented:
Sorry, I just noticed you removed the status filter. It should not impair performances that much.  Here is a comparison: https://gist.github.com/yvanzo/53c381842775f6cd053a68e42efd8a73 Can you put it back?

naiveaiguy commented:
@yvanzo Amended. Problem with my test before was that I added `AND status NOT IN (3, 4)` in the *subquery* rather than after the `FROM` block, where it works equally well.
